crates/pallet-domains/src/lib.rs

```diff
@@ -770,14 +747,35 @@ impl<T: Config> Pallet<T> {
                         Receipts::<T>::remove(receipt_hash);
                     }
                 }
-            })
+            });
+            OldestReceiptNumber::<T>::put(to_prune + One::one());
         }
 
         Self::deposit_event(Event::NewExecutionReceipt {
             primary_number,
             primary_hash,
         });
     }
+
+    fn apply_non_new_best_receipt(
+        execution_receipt: &ExecutionReceipt<T::BlockNumber, T::Hash, T::SecondaryHash>,
+    ) {
+        let primary_hash = execution_receipt.primary_hash;
+        let primary_number = execution_receipt.primary_number;
+        let receipt_hash = execution_receipt.hash();
+
+        // Track the fork receipt if it's not seen before.
```

Comment:
Hm... why are we tracking forks? This seems to be a new behavior.

More specifically given updated checks in `pre_dispatch_submit_bundle` I **think** it is now possible to leak runtime storage because this branch will potentially create entries below `OldestReceiptNumber` that will never be removed, unless I'm missing something.